src/py/pyodide/_package_loader.py

```diff
@@ -0,0 +1,84 @@
+from ._core import to_js, JsProxy
+
+import shutil
+import sysconfig
+import tarfile
+
+from pathlib import Path
+from site import getsitepackages
+from tempfile import NamedTemporaryFile
+from typing import List, IO, Iterable
+from zipfile import ZipFile
+
+SITE_PACKAGES = Path(getsitepackages()[0])
+STD_LIB = Path(sysconfig.get_path("stdlib"))  # type: ignore
+TARGETS = {"site": SITE_PACKAGES, "lib": STD_LIB}
+ZIP_TYPES = {".whl", ".zip"}
+TAR_TYPES = {".tar", ".gz", ".bz", ".gz", ".tgz", ".bz2", ".tbz2"}
+
+
+def unpack_buffer(filename: str, buffer: JsProxy, target: str = "site") -> JsProxy:
```

Comment:
filename as Path should also work I think?

Reply:
Yeah it would work, though we are calling this from Javascript so in practice it's gonna be a string.